src/geometry/LineUtil.js

```diff
@@ -240,3 +241,59 @@ export function _flat(latlngs) {
 	console.warn('Deprecated use of _flat, please use L.LineUtil.isFlat instead.');
 	return isFlat(latlngs);
 }
+
+/* @function polylineCenter(latlngs: LatLng[], map: Map): LatLng
+ * Returns the center of the `Polyline` of the passed LatLngs.
+ * The returned LatLng will be on the `Polyline`.
+ */
+export function polylineCenter(latlngs, map) {
+	var i, halfDist, segDist, dist, p1, p2, ratio, center;
+
+	if (!latlngs || latlngs.length === 0) {
+		throw new Error('latlngs not passed');
+	}
+
+	if (!isFlat(latlngs)) {
+		throw new Error('latlngs are not flat!');
+	}
+
+	// throws error when map is not passed because calculation requires projected coordinates
+	if (!map) {
+		throw new Error('map not passed');
+	}

```

Comment:
Why do we need explicit throw here?
It will throw anyway on next line.

Reply:
Copied from original `getCenter()`. It is clearer but it is not necessary